azkaban-common/src/main/java/azkaban/executor/container/ContainerizedDispatchImpl.java

```diff
@@ -0,0 +1,415 @@
+   * status for executions in queue is PREPARING. But for containerized dispatch method, the status
+   * will be READY.
+   * @return
+   */
+  @Override
+  public Status getStartStatus() {
+    return Status.READY;
+  }
+
+  /**
+   * This method will shutdown the queue processor thread. It won't pick up executions from queue
+   * for dispatch after this. This method will be called when webserver will shutdown.
+   */
+  @Override
+  public void shutdown() {
+    logger.info("Shutting down queue processor thread for containerized dispatch implementation.");
+    if (null != this.queueProcessor) {
+      this.queueProcessor.shutdown();
+    }
+  }
+
+  /**
+   * This method is used to enable queue processor thread. It will resume dispatching executions.
+   */
+  @Override
+  public void enableQueueProcessorThread() {
+    this.queueProcessor.setActive(true);
+  }
+
+  /**
+   * This method is used to disable queue processor thread. It will stop dispatching executions.
+   */
+  @Override
+  public void disableQueueProcessorThread() {
+    this.queueProcessor.setActive(false);
+  }
+
+  public State getQueueProcessorThreadState() {
+    return this.queueProcessor.getState();
+  }
+
+  public boolean isQueueProcessorThreadActive() {
+    return this.queueProcessor.isActive();
+  }
+
+  /**
+   * This QueueProcessorThread will fetch executions from database in batch/single execution at a
+   * time and dispatch it in container.
+   */
+  private class QueueProcessorThread extends Thread {
+
+    private final Props azkProps;
+    private volatile boolean shutdown = false;
+    private volatile boolean isActive = true;
+    private ExecutorService executorService;
+    private ExecutorLoader executorLoader;
+    private boolean executionsBatchProcessingEnabled;
+    private int executionsBatchSize;
+    private static final long QUEUE_PROCESSOR_WAIT_IN_MS = 1000;
+
+    public QueueProcessorThread(final Props azkProps, final ExecutorLoader executorLoader) {
+      this.azkProps = azkProps;
+      this.executorLoader = executorLoader;
+      setActive(
+          this.azkProps.getBoolean(Constants.ConfigurationKeys.QUEUEPROCESSING_ENABLED, true));
+      this.executionsBatchProcessingEnabled = azkProps
+          .getBoolean(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_ENABLED,
+              false);
+      this.executionsBatchSize =
+          azkProps
+              .getInt(ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_BATCH_SIZE,
+                  10);
+
+      this.executorService = Executors.newFixedThreadPool(azkProps.getInt(
+          ContainerizedExecutionManagerProperties.CONTAINERIZED_EXECUTION_PROCESSING_THREAD_POOL_SIZE,
+          10));
+      this.setName("Containerized-QueueProcessor-Thread");
+    }
+
+    @Override
+    public void run() {
+      // Loops till QueueProcessorThread is shutdown
+      while (!this.shutdown) {
+        synchronized (this) {
```

Comment:
Why is this synchronization needed ? Do we expect more than one thread to be running this?

Reply:
There is another method which is in abstract class to submit the flow. That will be running from different thread that is why.

Reply:
I have removed synchronize after testing it on Holdem4 to see if there is any issue happening because of submit flow method. Thanks!